src/components/generic/Alert.tsx

```diff
@@ -1,81 +1,154 @@
-import classes from './Alert.module.scss'
+import React, { createContext, useContext, useState, useEffect, useRef, ReactNode } from 'react';
+import { CheckCircle, Info, Warning, XCircle } from '@phosphor-icons/react';
+import classNames from 'classnames/bind';
+import classes from './Alert.module.scss';
 
-import {
-  CheckCircle,
-  Info,
-  Warning,
-  WarningCircle,
-} from '@phosphor-icons/react'
-import classNames from 'classnames/bind'
-import { useEffect, useRef } from 'react'
+const cx = classNames.bind(classes);
 
-const cx = classNames.bind(classes)
-
-export type SeverityType = 'error' | 'warning' | 'info' | 'success'
+export type SeverityType = "success" | "error" | "info" | "warning";
 
 interface AlertProps {
-  severity: SeverityType
-  description?: string
-  title?: string
-  marginTop?: boolean
-  marginButton?: boolean
-  className?: string
-  isScrollIntoView?: boolean
+  severity: SeverityType;
+  description?: string;
+  title?: string;
+  marginTop?: boolean;
+  className?: string;
+  style?: React.CSSProperties;
+  onDismiss?: () => void;
 }
 
 export type AlertType = {
-  severity: SeverityType
-  title: string
-  description: string
-}
+  id: string;
+  severity: SeverityType;
+  title: string;
+  description: string;
+};
+
+const AlertsContext = createContext<any>(undefined);
```

Comment:
Using `any` and `undefined` to initialize a context is not good practice.
Instead, I suggest doing this:
1. Define a `defaultState` object for the context.
2. Derive a type from that object using `type AlertsContextState = typeof defaultState`.
3. Define the context as `const AlertsContext = createContext<AlertsContextState>(defaultState)`